features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/internal/DatadogFlagsClient.kt

```diff
@@ -156,23 +168,60 @@ internal class DatadogFlagsClient(
         val precomputedFlag = flagsRepository.getPrecomputedFlag(flagKey)
         val convertedValue = precomputedFlag?.variationValue?.let(converter)
 
-        if (convertedValue != null && flagsConfiguration.trackExposures) {
-            writeExposureEvent(flagKey, precomputedFlag)
+        if (convertedValue == null) {
+            return defaultValue
         }
 
-        return convertedValue ?: defaultValue
-    }
+        val evaluationContext = flagsRepository.getEvaluationContext()
+        if (evaluationContext == null) {
+            /**
+             * this might happen if a previous session saved precomputed flags and a new session did not provide a valid context
+             */
+            featureSdkCore.internalLogger.log(
+                target = InternalLogger.Target.MAINTAINER,
+                level = InternalLogger.Level.ERROR,
+                messageBuilder = { ERROR_NO_EVALUATION_CONTEXT }
+            )
+        } else {
+            if (flagsConfiguration.trackExposures) {
+                writeExposureEvent(flagKey, precomputedFlag, evaluationContext)
+            }
+        }
 
-    private fun writeExposureEvent(name: String, data: PrecomputedFlag) {
-        val context = flagsRepository.getEvaluationContext()
-        if (context != null) {
-            featureSdkCore
-                .getFeature(FLAGS_FEATURE_NAME)
-                ?.unwrap<FlagsFeature>()?.processor?.processEvent(
-                    flagName = name,
-                    context = context,
-                    data = data
-                )
+        if (flagsConfiguration.rumIntegrationEnabled) {
+            logEvaluation(
+                key = flagKey,
+                value = convertedValue
+            )
         }
+
+        return convertedValue
+    }
+
+    private fun writeExposureEvent(name: String, data: PrecomputedFlag, context: EvaluationContext) {
+        featureSdkCore
+            .getFeature(FLAGS_FEATURE_NAME)
+            ?.unwrap<FlagsFeature>()?.processor?.processEvent(
```

Comment:
We cannot inject `processor` simply when we create `DatadogFlagsClient`? Not sure what is the point of `featureSdkCore.getFeature(FLAGS_FEATURE_NAME)?.unwrap<FlagsFeature>()` if it is the same module.